File: rms-client/rms/src/model/Request.ts
import {Sample} from "@/model/Sample";
import {Service} from "@/model/Service";
import {Report} from "@/model/Report"
import {User} from "@/model/User";
import {RequestRelation} from "@/model/RequestRelation";
import {RequestGroup} from "@/model/RequestGroup";

export interface Request {
    service?: Service
    user_service_id?: string
    status?: string
    memo?: string
    department?: string
    ward?: string
    physician?: string
    sample?: Sample
    create_at?: Date
    cart_at?: Date
    confirmed_at?: Date,
    lims_received_at?: Date,
    lims_resample_at?: Date,
    lims_completed_at?: Date,
    lims_resample_reason?: string,
    courier_company?: string
    awb_number?: string
    report?: Report
    reports?: Report[]
    user?: User
    request_relation?: RequestRelation
    request_group?: RequestGroup
}

File: cart-service/src/main/kotlin/com/gcgenome/rms/data/Request.kt
package com.gcgenome.rms.data

import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer
import java.time.LocalDateTime
import java.util.*

data class Request(
    var orderId: UUID?,
    var serviceId: String?,
    var sampleId: String?,
    var userServiceId: String?,
    var status: String?,
    var memo: String?,
    var department: String?,
    var ward: String?,
    var physician: String?,
    @JsonDeserialize(using = LocalDateTimeDeserializer::class)
    @JsonSerialize(using = LocalDateTimeSerializer::class)
    var createAt: LocalDateTime?,
    var sample: Sample?,
    var service: Service?
)